girder/models/upload.py

```diff
@@ -172,10 +175,13 @@ def finalizeUpload(self, upload, assetstore=None):
         self.remove(upload)
 
         # Add an async event for handlers that wish to process this file.
-        events.daemon.trigger('data.process', {
+        eventParams = {
             'file': file,
             'assetstore': assetstore
-        })
+        }
+        if 'reference' in upload:
+            eventParams['reference'] = upload['reference']
+        events.daemon.trigger('data.process', eventParams)
```

Comment:
Great.  I think the two PRs do fundamentally different things (both useful) -- this allows the initiator of a job run via girder_worker or romanesco to monitor the data.process events and determine if a file was the result of a particular job.